### PR TITLE
Squash docker image layers

### PR DESCRIPTION
**Pull request by L3n41c**
### What does this PR do?

Squash Datadog agent docker image layers.

### Motivation

As each Datadog agent release is based on the latest Debian base image, which is updated more often than Datadog agent releases, the docker images of two consecutive Datadog agent release are sharing no common layer.
So, the layering of the Datadog agent docker images is providing no gain.

### Additional Notes

I would have preferred to use `docker build --squash=true …` rather than a 3rd party tool but the docker daemon of the GitLab CI doesn’t have experimental features enabled.

### Describe your test plan

Check that the Datadog docker images is mono-layer with `docker history`.

### CODE REVIEW

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -2658,7 +2658,9 @@ send_pkg_size-a7:
     # Build testing stage if provided
     - test "$TESTING_ARG" && docker build --file $BUILD_CONTEXT/$ARCH/Dockerfile $TESTING_ARG $BUILD_CONTEXT
     # Build release stage and push to ECR
-    - docker build $BUILD_ARG --file $BUILD_CONTEXT/$ARCH/Dockerfile --pull --tag $TARGET_TAG $BUILD_CONTEXT
+    - docker build $BUILD_ARG --file $BUILD_CONTEXT/$ARCH/Dockerfile --pull --tag ${TARGET_TAG}-unsquashed $BUILD_CONTEXT
+    - pip install docker-squash
```
Comment (vboulineau) [inline .gitlab-ci.yml:2662]: I think the trick when experimental is not enabled is to use `docker export | docker import`

Comment (L3n41c) [reply .gitlab-ci.yml:2662]: Yeah, but doing this looses all the image metadata like `CMD`, `ENV`, etc. It creates a new mono-layer image with the same filesystem but all `CMD`, `ENV`, `VOLUMES`, etc. are lost.
And we need at least `CMD` to target S6 and some environment variables.

Comment (vboulineau) [reply .gitlab-ci.yml:2662]: Ah okay, my bad then. The only other option I see is to use `docker buildx build --squash` which requires experimental on client side (not sure if it also requires experimental on daemon side, not documented at least).

Comment (L3n41c) [reply .gitlab-ci.yml]: Thanks for the tip @vboulineau !
I’ve switched to `docker buildx build --squash`. It works and it doesn’t require any 3rd party tool.

Comment (L3n41c) [reply .gitlab-ci.yml]: `docker buildx build --squash` isn’t implemented:
https://github.com/docker/buildx/blob/master/commands/build.go#L76

**Comment by L3n41c** [comment]
* `docker build --squash` doesn’t work because it requires to enable the experimental features on the daemon side.
* `docker buildx build --squash` doesn’t work because it is not implemented.
* `docker export | docker import` doesn’t work because it looses all the metadata like `CMD`, `ENV`, `VOLUMES`, etc. It keeps only the filesystem content.

**Review by derekwbrown — APPROVED** [review]

**Review by xornivore — APPROVED** [review]
❤️

**Pull request merged**